homeassistant/components/sonos/speaker.py

```diff
@@ -283,25 +283,23 @@ def share_link(self) -> ShareLinkPlugin:
         return self._share_link_plugin
 
     @property
-    def subscription_address(self) -> str | None:
+    def subscription_address(self) -> str:
         """Return the current subscription callback address if any."""
-        if self._subscriptions:
-            addr, port = self._subscriptions[0].event_listener.address
-            return ":".join([addr, str(port)])
-        return None
+        addr, port = self._subscriptions[0].event_listener.address
```

Comment:
```suggestion
        assert len(self._subscriptions) > 0
        addr, port = self._subscriptions[0].event_listener.address
```
That could fail if `self._subscriptions` is empty. Might be worth adding an `assert` just to be safe (and have a slightly better error message).